src/Parsing/Impl/Ast/INode.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Python.Core.Text;
+
+namespace Microsoft.Python.Parsing.Ast {
+    public interface INode {
+        int EndIndex { get; set; }
+
+        int StartIndex { get; set; }
+
+        IEnumerable<Node> GetChildNodes();
+        void Walk(PythonWalker walker);
+        Task WalkAsync(PythonWalkerAsync walker, CancellationToken cancellationToken = default);
+
+        string NodeName { get; }
+
+        string ToCodeString(PythonAst ast);
```

Comment:
I would drop `ToCodeString` from public. It is buggy and we'd rather remove it eventually.

Reply:
Yes!